CMakeLists.txt

```diff
@@ -578,15 +574,72 @@ if(SWITCH)
   build_switch_binaries(${BIN_TARGET})
 endif()
 
-if(${CMAKE_SYSTEM_NAME} STREQUAL Linux)
-  set(CPACK_PACKAGE_NAME "devilutionx")
-  string(TOLOWER "${CPACK_PACKAGE_NAME}-${CMAKE_SYSTEM_NAME}-${CMAKE_SYSTEM_PROCESSOR}" CPACK_PACKAGE_FILE_NAME)
-  set(CPACK_STRIP_FILES TRUE)
-  set(CPACK_GENERATOR "7Z")
+if(CPACK)
+  if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    string(TOLOWER ${PROJECT_NAME} project_name)
+    set(CPACK_PACKAGE_NAME ${project_name})
+    set(CPACK_GENERATOR "7Z")
+    # Common *nix files
+    set(CPACK_STRIP_FILES TRUE)
+    install(TARGETS ${BIN_TARGET} DESTINATION bin)
+    set(desktop_file "${PROJECT_SOURCE_DIR}/Packaging/fedora/${project_name}.desktop")
+
+    find_program(DFI desktop-file-install)
+    if(DFI)
+      execute_process(COMMAND ${DFI} --dir=${CMAKE_BINARY_DIR} ${desktop_file})
+      set(desktop_file "${CMAKE_BINARY_DIR}/${project_name}.desktop")
+    endif()
 
-  install (TARGETS ${BIN_TARGET} DESTINATION ./)
-  install (FILES "${PROJECT_SOURCE_DIR}/Packaging/resources/CharisSILB.ttf" DESTINATION ./)
-  install (FILES "${PROJECT_SOURCE_DIR}/Packaging/resources/LICENSE.CharisSILB.txt" DESTINATION ./)
-  install (FILES "${PROJECT_SOURCE_DIR}/Packaging/nix/README.txt" DESTINATION ./)
+    install(FILES "${desktop_file}"
+      DESTINATION "share/applications"
+    )
+    install(FILES
+        "${PROJECT_SOURCE_DIR}/Packaging/nix/README.txt"
+        "${PROJECT_SOURCE_DIR}/Packaging/resources/LICENSE.CharisSILB.txt"
+      DESTINATION "share/diasurgical/${project_name}"
+    )
+    install(FILES "${PROJECT_SOURCE_DIR}/Packaging/resources/16.png"
+      DESTINATION "share/icons/hicolor/16x16/apps"
+      RENAME "${project_name}.png"
+    )
+    install(FILES "${PROJECT_SOURCE_DIR}/Packaging/resources/Diablo_32.png"
+      DESTINATION "share/icons/hicolor/32x32/apps"
+      RENAME "${project_name}.png"
+    )
+    install(FILES "${PROJECT_SOURCE_DIR}/Packaging/resources/Diablo_48.png"
+      DESTINATION "share/icons/hicolor/48x48/apps"
+      RENAME "${project_name}.png"
+    )
+    install(FILES "${PROJECT_SOURCE_DIR}/Packaging/resources/CharisSILB.ttf"
+      DESTINATION "share/fonts/truetype"

```

Comment:
the game doesn't look for the font here, are you sure this will work?

Reply:
On our own branch for Amiga, we have added `set(TTF_FONT_PATH \"LiberationSerif-Bold.ttf\")`

Shouldn't it be possible to check if target == package change font-path to /usr/share/fonts/truetype etc?

Reply:
I was emulating what the fedora rpm was doing. Where does the source look for it currently? Seems like a small-ish change for packaging to match the code, but seems like a more expected behavioral change for the code to look for it in a "standard" location.

Reply:
ahhh, the fedora spec file has this line:
`export CXXFLAGS="-DTTF_FONT_PATH=\"/usr/share/fonts/truetype/CharisSILB.ttf\""`
I missed that.

Reply:
I pushed a small patch that splits `TTF_FONT_PATH` into `TTF_FONT_NAME` and `TTF_FONT_DIR` so I could search for `TTF_FONT_NAME` in 3 places:
1) Current directory
2) `TTF_FONT_DIR`
3) `#ifdef __linux__` "/usr/share/fonts/truetype"
It should preserve the current behavior while allowing the default build on Linux to search for the `TTF_FONT_NAME` in a known good location.

Reply:
The font is used in the credits and any error message.

Your patch looks like a decent solution.